pkg/apis/network/v1/pki_types.go

```diff
@@ -0,0 +1,60 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PKISpec defines the desired state of PKI.
+// +k8s:openapi-gen=true
+type PKISpec struct {
+	// caCert is information about the CA certificate and key.
+	CACert CertSpec `json:"caCert"`
+
+	// caBundleName is the name of the config-map containing the CA bundle.
+	CABundleName string `json:"caBundleName"`
+
+	// serverCert is the server certificate to create.
+	// It will be created with both ServerAuth and ClientAuth
+	ServerCert CertSpec `json:"serverCert"`
+}
+
+// CertSpec is generic information about certificates
+type CertSpec struct {
+	// name is the CN in the certificate as well as the name of the Secret
+	// that will be generated.
+	Name string `json:"name"`
+
+	// validFor is the desired length for certificate validity.
+	// Certificates will be rotated at 1/2 the length of their validity.
+	// for example, "720h"
+	ValidFor metav1.Duration `json:"validFor"`
+}
+
+// PKIStatus defines the observed state of PKI
+// +k8s:openapi-gen=true
+type PKIStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PKI is a simple certificate authority. It creates a CA and generates
+// certificates signed by that CA.
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=pkis,scope=Namespaced
+type PKI struct {
```

Comment:
This type should come first in the file. Also, the docs make it sound like the API object itself is taking action, rather than just describing the actions that the CNO will take.